agent/api/container.go

```diff
@@ -310,6 +310,24 @@ func (c *Container) SetRegistryAuthCredentials(credential credentials.IAMRoleCre
 	c.RegistryAuthentication.ECRAuthData.SetPullCredentials(credential)
 }
 
+// IsASMImagePullAuth returns true if the acs payload contains a reference the
+// image pull auth data stored in AWS Secrets Manager
+func (c *Container) IsASMImagePullAuth() bool {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	return c.RegistryAuthentication != nil &&
+		c.RegistryAuthentication.Type == "asm" &&
+		c.RegistryAuthentication.ASMAuthData != nil
```

Comment:
i was doing this check since right after we tag the `ASMAuthData` with the associated task arn, so this is where we reach into `ASMAuthData` and `SetTaskARN(...)`.